src/lib.rs

```diff
@@ -776,60 +870,103 @@ fn split_flags(output: &[u8]) -> Vec<String> {
-        )]
-    ));
-    assert!(!is_static_available(
-        "ffi_convenience",
-        &system_roots,
-        &[PathBuf::from(
-            "/Library/Ruby/Gems/2.0.0/gems/ffi-1.9.10/ext/ffi_c/libffi-x86_64/.libs"
-        )]
-    ));
-
-    // Homebrew is in /usr/local, and it's not a part of the OS
-    if Path::new("/usr/local/lib/libpng16.a").exists() {
-        assert!(is_static_available(
-            "png16",
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    #[cfg(target_os = "macos")]
+    fn system_library_mac_test() {
+        use std::path::Path;
+
+        let system_roots = vec![PathBuf::from("/Library"), PathBuf::from("/System")];
+
+        assert!(!is_static_available(
+            "PluginManager",
             &system_roots,
-            &[PathBuf::from("/usr/local/lib")]
+            &[PathBuf::from("/Library/Frameworks")]
         ));
+        assert!(!is_static_available(
+            "python2.7",
+            &system_roots,
+            &[PathBuf::from(
+                "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config"
+            )]
+        ));
+        assert!(!is_static_available(
+            "ffi_convenience",
+            &system_roots,
+            &[PathBuf::from(
+                "/Library/Ruby/Gems/2.0.0/gems/ffi-1.9.10/ext/ffi_c/libffi-x86_64/.libs"
+            )]
+        ));
+
+        // Homebrew is in /usr/local, and it's not a part of the OS
+        if Path::new("/usr/local/lib/libpng16.a").exists() {
+            assert!(is_static_available(
+                "png16",
+                &system_roots,
+                &[PathBuf::from("/usr/local/lib")]
+            ));
+
+            let libpng = Config::new()
+                .range_version("1".."99")
+                .probe("libpng16")
+                .unwrap();
+            assert!(libpng.version.find('\n').is_none());
+        }
+    }
 
-        let libpng = Config::new()
-            .range_version("1".."99")
-            .probe("libpng16")
-            .unwrap();
-        assert!(libpng.version.find('\n').is_none());
+    #[test]
+    #[cfg(target_os = "linux")]
+    fn system_library_linux_test() {
+        assert!(!is_static_available(
+            "util",
+            &[PathBuf::from("/usr")],
+            &[PathBuf::from("/usr/lib/x86_64-linux-gnu")]
+        ));
+        assert!(!is_static_available(
+            "dialog",
+            &[PathBuf::from("/usr")],
+            &[PathBuf::from("/usr/lib")]
+        ));
     }
-}
 
-#[test]
-#[cfg(target_os = "linux")]
-fn system_library_linux_test() {
-    assert!(!is_static_available(
-        "util",
-        &[PathBuf::from("/usr")],
-        &[PathBuf::from("/usr/lib/x86_64-linux-gnu")]
-    ));
-    assert!(!is_static_available(
-        "dialog",
-        &[PathBuf::from("/usr")],
-        &[PathBuf::from("/usr/lib")]
-    ));
+    fn test_library_filename(target: &str, filename: &str) {
```

Comment:
micronit: mind adding `#[track_caller]` here so that if it fails it's more clear which check is the issue?

Reply:
done

Reply:
That's stable in 1.46, not 1.30. Would need to bump the MSRV for that.

Reply:
whoops, reverted

Reply:
@thomcc All good for you, or would you prefer bumping MSRV to 1.46 just for the test?

Reply:
Stuff in a disabled `cfg(...)` wouldn't impact the MSRV of downstream consumers FWIW. Otherwise rust version checks in build.rs and the like wouldn't matter. This would complicate the ability to check MSRV in CI, though (and the `#[track_caller]` is probably not beneficial enough to warrant much complexity).

I tend not to think MSRV is very important for us to worry about (as I've stated in https://github.com/rust-lang/libs-team/issues/72), but if you'd rather wait it doesn't bother me.

Reply:
Personally I don't think MSRV is something worth worrying about, but some loud people tend to care a lot so... :)

I'm fine with it either way. We could also keep it used in the test and just not run the tests on the 1.30 CI.

Reply:
I generally don't think MSRV is very important to worry about, however, this is a very widely used crate, and the benefit of increasing the MSRV is very small, so I think it's better to not increase it in this case.

Reply:
This wouldn't actually increase the MSRV to use the crate, it would just increase the MSRV needed to run the tests. Downstream customers don't need to run the tests, only use the crate, so it doesn't really matter.

That said I agree that the benefit isn't worth it here, for now.

Reply:
I've merged it.